components/CardForm/index.js

```diff
@@ -55,11 +84,14 @@ export default function CardForm({
     }
   }
 
-  const inputRef = useRef(null);
-
   useEffect(() => {
     if (existingActivityData) {
       inputRef.current.focus();
+
+      setSelectedDate(
+        existingActivityData.date ? new Date(existingActivityData.date) : null
+      );
+      setSelectedCategory(existingActivityData.category || "");
```

Comment:
We could potentially tidy this up with some destructuring. 

```suggestion
      const { date, category } = existingActivityData;
      setSelectedDate(date ? new Date(date) : null);
      setSelectedCategory(category || "");
```

Reply:
Additional thought. It seems like you are using `existingActivityData` throughout the component. So we could move this declaration out of the useEffect so that it can be used everywhere.